tests/indexes_/test_atlas_indexes.py

```diff
@@ -0,0 +1,194 @@
+    def test_simple(self):
+        with connection.schema_editor() as editor:
+            index = SearchIndex(
+                name="recent_article_idx",
+                fields=["number"],
+            )
+            editor.add_index(index=index, model=Article)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+    @skipIfDBFeature("supports_atlas_search")
+    def test_index_not_created(self):
+        with connection.schema_editor() as editor:
+            index = SearchIndex(
+                name="recent_article_idx",
+                fields=["number"],
+            )
+            editor.add_index(index=index, model=Article)
+            self.assertNotIn(
+                index.name,
+                connection.introspection.get_constraints(
+                    cursor=None,
+                    table_name=Article._meta.db_table,
+                ),
+            )
+
+    @skipUnlessDBFeature("supports_atlas_search")
+    def test_multiple_fields(self):
+        with connection.schema_editor() as editor:
+            index = SearchIndex(
+                name="recent_article_idx",
+                fields=["headline", "number", "body", "data", "embedded", "created_at"],
+            )
+            editor.add_index(index=index, model=Article)
+            index_info = connection.introspection.get_constraints(
+                cursor=None,
+                table_name=Article._meta.db_table,
+            )
+            expected_options = {
+                "dynamic": False,
+                "fields": {
+                    "created_at": {"type": "date"},
+                    "body": {
+                        "indexOptions": "offsets",
+                        "norms": "include",
+                        "store": True,
+                        "type": "string",
+                    },
+                    "data": {"dynamic": False, "fields": {}, "type": "document"},
+                    "embedded": {"dynamic": False, "fields": {}, "type": "embeddedDocuments"},
+                    "headline": {
+                        "indexOptions": "offsets",
+                        "norms": "include",
+                        "store": True,
+                        "type": "string",
+                    },
+                    "number": {
+                        "indexDoubles": True,
+                        "indexIntegers": True,
+                        "representation": "double",
+                        "type": "number",
+                    },
+                },
+            }
+            self.assertCountEqual(index_info[index.name]["columns"], index.fields)
+            self.assertEqual(index_info[index.name]["options"], expected_options)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+
+class VectorSearchIndexTests(TestMixin, TestCase):
+    """
+    Tests for creating, validating, and removing vector search indexes
+    using Django's schema editor.
+    """
+
+    def test_invalid_similarity_function(self):
```

Comment:
Some tests like this one and deconstruct can be moved to a `SimpleTestCase` which is a little faster and doesn't need the db skip.